Tests/SwiftAlgorithmsTests/WindowsTests.swift

```diff
@@ -0,0 +1,88 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift Algorithms open source project
+//
+// Copyright (c) 2020 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+//
+//===----------------------------------------------------------------------===//
+
+import XCTest
+import Algorithms
+
+final class WindowsTests: XCTestCase {
+  
+  func testWindowsOfString() {
+    let s = "swift"
+    let w = s.windows(size: 2)
+    var i = w.startIndex
+    
+    XCTAssertEqualSequences(w[i], "sw")
+    w.formIndex(after: &i)
+    XCTAssertEqualSequences(w[i], "wi")
+    w.formIndex(after: &i)
+    XCTAssertEqualSequences(w[i], "if")
+    w.formIndex(after: &i)
+    XCTAssertEqualSequences(w[i], "ft")
+    w.formIndex(after: &i)
+    XCTAssertTrue(w[i].isEmpty)
```

Comment:
`Array` is kind of singular, in that it's built so that those index offsets optimize down to simple additions and subtractions. Most of the rest of our standard library collections (like `String`, `Dictionary`, and `Set`) will trap if you try to move an index out of bounds, to enforce the preconditions on the `Collection` protocol requirements.